src/graph/serializers/decoders/prev/v4/decode_graph.c

```diff
@@ -5,21 +5,51 @@
 */
 
 #include <assert.h>
-#include "prev_decode_graph.h"
-#include "../../../graph.h"
+#include "decode_v4.h"
+
+typedef enum {
+	V4_T_NULL = 0,
+	V4_T_STRING = 0x001,
+	V4_T_INT64 = 0x004,
+	V4_T_BOOL = 0x010, // shares 'longval' representation in SIValue union
+	V4_T_DOUBLE = 0x040,
+	V4_T_PTR = 0x080,
+	V4_T_CONSTSTRING = 0x100, // only used in deserialization routine
+	V4_T_NODE = 0x200,
+	V4_T_EDGE = 0x400,
+} PrevSIType;
```

Comment:
Maybe move this typedef to a different file, to be used here and in v_5. same values

Reply:
In this case I feel like simpler and more modular version folders are better than less redundancy, but am willing to discuss reusing this and the associated routine.